test/games/strategy/triplea/delegate/MockObjects.java

```diff
@@ -83,19 +83,37 @@ public Unit answer(InvocationOnMock invocation) throws Throwable {
   }
 
   public static IServerMessenger getDummyMessenger() {
-    IServerMessenger messenger = null;
-    try {
-      messenger = spy(new ServerMessenger("", 0){
-        @Override
-        public void removeConnection(INode node){
-          for (final IConnectionChangeListener listener : connectionListeners) {
-            listener.connectionRemoved(node);
-          }
+    IServerMessenger messenger = mock(IServerMessenger.class);
```

Comment:
Points for creativity. The way I see to get rid of the dummy messenger is to remove calls to `public static IServerMessenger getDummyMessenger() {` altogether, and then remove the method. The upstream code would need to deal with cases where there is no server messenger object. Likely the abstractions are not quite right, and too much coupling means we wind up creating a dummy version of this object, even though we could skip unnecessary computation.